pkg/estimator/client/general.go

```diff
@@ -228,3 +229,45 @@ func getMaximumReplicasBasedOnResourceModels(cluster *clusterv1alpha1.Cluster, r
 
 	return maximumReplicasForResource, nil
 }
+
+// CalcAvailableReplicas get the minimum value of MaxAvailableReplicas in terms of all estimators.
+func CalcAvailableReplicas(clusters []*clusterv1alpha1.Cluster, spec *workv1alpha2.ResourceBindingSpec) []workv1alpha2.TargetCluster {
```

Comment:
I think this method should be placed in the `pkg/util` directory or somewhere else. For the estimator, all callers should call the interface directly.